File: messages/AddATCMessage.js
const IMessage = require('./IMessage')

module.exports = class AddATCMessage extends IMessage {
  command = '#AA'

  constructor(source, destination, realName, account, password, rating, protocolVersion = 'B') {
    super()
    this.source = source
    this.destination = destination
    this.realName = realName
    this.account = account
    this.password = password
    this.rating = rating
    this.protocolVersion = protocolVersion
  }

  parseRawMessage(rawMessage) {
    const [_command, source, destination, realName, account, password, rating, protocolVersion] = rawMessage.split(':')
    return new AddATCMessage(source, destination, realName, account, password, rating, protocolVersion)
  }

  toString() {
    return this.command + [
      this.source,
      this.destination,
      this.realName,
      this.account,
      this.password,
      this.rating,
      this.protocolVersion
    ].join(':')
  }
}

File: messages/AddATCMessage.py
from messages.IMessage import IMessage


class AddATCMessage(IMessage):
    command = '#AA'

    def __init__(
        self,
        source: str,
        destination: str,
        real_name: str,
        account: str,
        password: str,
        rating: int,
        protocol_version: str = 'B'
    ):
        super().__init__()
        self.source = source
        self.destination = destination
        self.real_name = real_name
        self.account = account
        self.password = password
        self.rating = rating
        self.protocol_version = protocol_version

    @staticmethod
    def parse_raw_message(raw_message):
        [source, destination, real_name, account, password,
            rating, protocol_version] = raw_message[3:].split(':')
        return AddATCMessage(source, destination, real_name, account, password, rating, protocol_version)

    def __str__(self):
        return self.command + ':'.join([
            self.source,
            self.destination,
            self.real_name,
            self.account,
            self.password,
            str(self.rating),
            self.protocol_version
        ])
